Content.Server/GameObjects/Components/Power/PowerCellComponent.cs

```diff
@@ -1,18 +1,39 @@
 ﻿using Content.Shared.GameObjects.Components.Power;
+using Content.Shared.GameObjects.EntitySystems;
 using Robust.Server.GameObjects;
 using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+using Robust.Shared.ViewVariables;
 
 namespace Content.Server.GameObjects.Components.Power
 {
     /// <summary>
-    ///     Batteries that have update an <see cref="AppearanceComponent"/> based on their charge percent.
+    /// Batteries that can update an <see cref="AppearanceComponent"/> based on their charge percent
+    /// and fit into a <see cref="PowerCellSlotComponent"/> of the appropriate size.
     /// </summary>
     [RegisterComponent]
     [ComponentReference(typeof(BatteryComponent))]
-    public class PowerCellComponent : BatteryComponent
+    public class PowerCellComponent : BatteryComponent, IExamine
     {
         public override string Name => "PowerCell";
 
+        [ViewVariables] public PowerCellSize CellSize => _cellSize;
+        private PowerCellSize _cellSize = PowerCellSize.Small;
+
+        /// <summary>
+        /// False if we shouldn't waste time updating the AppearanceComponent (if eg. the entity doesn't have one).
+        /// </summary>
+        private bool _updateVisual = true;
```

Comment:
Not used, also can't they just call TryGetComponent for an appearance component anyway).

Reply:
I figured it'd save some cycles to just check this flag if we know it's a simple battery instead of doing a method call, but that's probably premature optimization.